aws/resource_aws_cloudfront_function.go

```diff
@@ -0,0 +1,303 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsCloudFrontFunction() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudFrontFunctionCreate,
+		Read:   resourceAwsCloudFrontFunctionRead,
+		Update: resourceAwsCloudFrontFunctionUpdate,
+		Delete: resourceAwsCloudFrontFunctionDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("name", d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
+		},
```

Comment:
By setting `name` in `resourceAwsCloudFrontFunctionRead()` we can use the pass-through implementation:

```suggestion
		Importer: &schema.ResourceImporter{
			State: schema.ImportStatePassthrough,
		},
```